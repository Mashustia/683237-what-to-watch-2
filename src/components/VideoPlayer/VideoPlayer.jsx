import React, {Component} from 'react';
import PropTypes from 'prop-types';

class VideoPlayer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isPlayed: false
    };

    this.videoRef = React.createRef();
    this.timer = null;
    this.onStopVideo = this.onStopVideo.bind(this);
    this.onPlayVideo = this.onPlayVideo.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.active !== this.props.active && !this.props.active) {
      this.onStopVideo();
    }

    if (prevProps.active !== this.props.active && this.props.active) {
      this.onPlayVideo();
    }
  }

  onStopVideo() {
    this.setState({
      isPlayed: false
    }, () => {
      const video = this.videoRef.current;
      clearTimeout(this.timer);
      video.currentTime = 0;
      video.pause();
    });
  }

  onPlayVideo() {
    const video = this.videoRef.current;
    this.timer = setTimeout(() => {
      this.setState({isPlayed: true}, () => video.play());
    }, 1000);
  }

  render() {
    const {preview} = this.props;

    return (
      <div className={this.state.isPlayed ? `film-active` : `film-hide`}>
        <video
          className={`preview`}
          src={preview}
          ref={this.videoRef}
          controls
          muted
        />
      </div>
    );
  }
}

VideoPlayer.propTypes = {
  preview: PropTypes.string,
  active: PropTypes.bool
};

export default VideoPlayer;
