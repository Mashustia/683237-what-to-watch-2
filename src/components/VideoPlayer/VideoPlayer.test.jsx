import React from 'react';
import renderer from 'react-test-renderer';

import VideoPlayer from './VideoPlayer';

it(`VideoPlayer component renders correctly`, () => {
  const component = renderer
    .create(<VideoPlayer
      preview={`test`}
      active={false}
    />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
