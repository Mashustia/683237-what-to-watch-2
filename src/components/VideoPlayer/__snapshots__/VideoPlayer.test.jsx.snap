// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer component renders correctly 1`] = `
<div
  className="film-hide"
>
  <video
    className="preview"
    controls={true}
    muted={true}
    src="test"
  />
</div>
`;
